fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java

```diff
@@ -233,6 +236,24 @@ public void setDopEstimated() {
         dopEstimated = true;
     }
 
+    public void computeLocalRfWaitingSet(PlanNode root, boolean clearGlobalRuntimeFilter) {
+        root.fillLocalRfWaitingSet(hashJoinNodeIds);
+        if (root instanceof HashJoinNode) {
+            hashJoinNodeIds.add(root.getId().asInt());
+        }
+        if (clearGlobalRuntimeFilter) {
+            root.clearProbeRuntimeFilters();
+            if (root instanceof HashJoinNode) {
+                ((HashJoinNode) root).clearBuildRuntimeFilters();
+            }
+        }
+        for (PlanNode child : root.getChildren()) {
+            if (child.getFragment() == this) {
+                computeLocalRfWaitingSet(child, clearGlobalRuntimeFilter);
+            }
```

Comment:
Maybe if `(child.getFragment() != this)`?

Reply:
child.getFragment() == this means only tackle child planNode in the same fragment instance.  here is collecting the local runtime filter.